src/librustc_trans/mir/block.rs

```diff
@@ -115,6 +115,9 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
                 if let Some(cleanup_pad) = cleanup_pad {
                     bcx.cleanup_ret(cleanup_pad, None);
                 } else {
+                    let llpersonality = bcx.fcx.eh_personality();
```

Comment:
Why is `init_cpad` not doing enough? You are not supposed to have `resume` without some sort of cleanups going on.